src/main/java/uk/ac/cam/cl/dtg/isaac/api/services/EmailService.java

```diff
@@ -91,29 +110,39 @@ public void sendAssignmentEmailToGroup(final IAssignmentLike assignment, final H
                 .putAll(tokenToValueMapping)
                 .build();
 
-            sendTemplatedEmailToActiveGroupMembers(userGroupDTO, templateName, map, EmailType.ASSIGNMENTS);
+            // Get email template
+            EmailTemplateDTO emailTemplate = emailManager.getEmailTemplateDTO(templateName);
+
+            if (this.userInMailGunBetaList(assignmentOwnerDTO)) {
+                mailGunEmailManager.sendBatchEmails(
+                        groupManager.getUsersInGroup(userGroupDTO),
+                        emailTemplate,
+                        EmailType.ASSIGNMENTS,
+                        Constants.IsaacMailGunTemplate.ASSIGNMENT,
+                        variables,
+                        null
+                );
+            } else {
+                // If user is not in the MailGun assignment emails beta list, use our standard email method
+                Map<Long, GroupMembershipDTO> userMembershipMapforGroup = this.groupManager.getUserMembershipMapForGroup(userGroupDTO.getId());
+                groupManager.getUsersInGroup(userGroupDTO).stream()
+                        // filter users so those who are inactive in the group aren't emailed
+                        .filter(user -> GroupMembershipStatus.ACTIVE.equals(userMembershipMapforGroup.get(user.getId()).getStatus()))
+                        // send an assignment email to each active user
+                        .forEach(user -> {
+                            try {
+                                emailManager.sendTemplatedEmailToUser(user, emailTemplate, variables, EmailType.ASSIGNMENTS);
+                            } catch (ContentManagerException | SegueDatabaseException e) {
```

Comment:
Yep this is because streams don't handle exceptions thrown in the given lambda functions. I think it's fine that it's more fine-grained error handling - there _could_ be some edge cases where the email sending fails for only one user, although I can't imagine how.